src/general/StageHUDBase.cs

```diff
@@ -716,6 +736,52 @@ public void UpdateEnvironmentalBars(BiomeConditions biome)
         // TODO: pressure?
     }
 
+    /// <summary>
+    ///   Creates and displays a fossilisation button above each on-screen microbe.
+    /// </summary>
+    public void ShowFossilisationButtons()
+    {
+        var microbes = GetTree().GetNodesInGroup(Constants.AI_TAG_MICROBE).Cast<Microbe>();
+        foreach (var microbe in microbes)
+        {
+            if (microbe.Species is not MicrobeSpecies)
+                continue;
+
+            var button = FossilisationButtonScene.Instance<FossilisationButton>();
+            button.AttachedMicrobe = microbe;
+            button.Connect(nameof(FossilisationButton.OnFossilisationDialogOpened), this,
+                nameof(ShowFossilisationDialog));
+
+            // Display a faded button with a different hint if the species has been fossilised.
+            var alreadyFossilised = FossilisedSpecies.CreateListOfSaves().Contains(
```

Comment:
Calling CreateListOfSaves in this loop sounds like a really bad idea to me... would be better to build that list outside this loop.